### DISCUSSION TITLE
developing a pool w/o official pool protocol?

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#5922] [Ideas]
This might be too late. But in principle, one can develop a pool just like the hpool w/o official pool protocol. The pool would only have one official wallet address to the external network, run a fast timelord, and stores the block chain copy, while participants do not store the blockchain, and only need to keep their plots online for the pool main node to fetch. The pool would then distribute the rewards to the wallets of the participants. This would also eliminate the sync problem had this been the direction from the beginning.

**Comment by padik11** [comment]
I am surprised that the pool protocol was not introduced from the beginning

**Reply by room101-dev** [reply]
It's quite simple, take a look at bitcoin, and how they do get-new-template() this RPC call is issued by the pool or solo-miner, they get a raw template for a new block, complete the templates with transactions, and then submit the template.

Yes, why in the hell didn't chia do this? Well its easy, because they wanted 100% control of the chia-net

So anyway, just study how bitcoin places this RPC, and add the source to CHIA, I'm sure this is how hpool did it, most of chia is just a clone of bitcoin like everything else.

Hell yes people, do it yourself, wait for chia-net, you will die in poverty of old age.

Run a tcp-ip packet sniffer on hpool client while your mining your blocks, reverse engineer what they're doing ,its quite easy, we do it all the time.

**Comment by padik11** [comment]
Could you advise on how to get started using hpool? They have instructions, but it's not entirely clear to me.
